### ISSUE TITLE
BTAG stuff (was BTagWeight 0.7869504?)

### ISSUE THREAD

**Issue by jpata** [vhbb/cmssw#147]
There was an issue described by @degrutto in an e-mail about the b-tag weight being very low for jets with pt>160.

I'm taking a look if there is no bug in the handling of the binning in the BTagWeightCalculator.
https://github.com/vhbb/cmssw/blob/vhbbHeppy74X/PhysicsTools/Heppy/python/physicsutils/BTagWeightCalculator.py

~~~
root [3] tree->Scan("Jet_mcFlavour:Jet_pt:Jet_eta:Jet_bTagWeight", "Jet_bTagWeight < 0.9")
***********************************************************************
*    Row   * Instance * Jet_mcFla *    Jet_pt *   Jet_eta * Jet_bTagW *
***********************************************************************
*     1105 *        1 *        -5 * 302.24673 * -0.858050 * 0.7869504 *
*     1145 *        0 *        -5 * 176.16626 * 0.5810403 * 0.7869504 *
*     1228 *        0 *        -5 * 246.33569 * 0.3852572 * 0.7869504 *
*     3385 *        0 *        -5 * 182.76608 * 1.4541995 * 0.7869504 *
*     3943 *        0 *         5 * 535.70007 * 0.8819748 * 0.7869504 *
*     9021 *        2 *         0 * 184.10034 * -0.684203 * 0.7869504 *
*    12306 *        0 *         5 * 265.02423 * -0.909397 * 0.7869504 *
*    12554 *        4 *        -5 * 213.87921 * 1.3983007 * 0.7869504 *
~~~

**Comment by arizzi** [comment]
any reason to believe 0.7869... is too low? (what surprise me is that it is flavour independent...)

**Comment by arizzi** [comment]
we should also have a look at
https://twiki.cern.ch/twiki/bin/view/CMS/BtagRecommendation74X50ns

**Comment by jpata** [comment]
Also to note:

because sometimes the input scale factors are negative, the per-jet weight turns out to be negative. Checking if this also happens with the updated weights (27.07)

![screen shot 2015-09-07 at 12 57 36](https://cloud.githubusercontent.com/assets/69717/9716072/a9ede0e4-5566-11e5-857a-651425a5a7de.png)

**Comment by arizzi** [comment]
mhh negative? does it makes sense?

**Comment by jpata** [comment]
It was a bug in the scale factor files. Fix in #157. I ran over tt+jets and tt+H with crab and didn't see these issues any more.

**Comment by jpata** [comment]
I would say it's OK to close.
